openhcl/virt_mshv_vtl/src/processor/snp/mod.rs

```diff
@@ -2352,8 +2353,13 @@ impl<T: CpuIo> UhHypercallHandler<'_, '_, T, SnpBacked> {
         self.vp.partition.hcl.tlbsync();
     }
 
-    fn do_flush_virtual_address_space(&mut self, processor_set: &[u32], flags: HvFlushFlags) {
-        let only_self = processor_set.len() == 1 && processor_set[0] == self.vp.vp_index().index();
+    fn do_flush_virtual_address_space(
+        &mut self,
+        processor_set: &ProcessorSet<'_>,
+        flags: HvFlushFlags,
+    ) {
+        let only_self = processor_set.len() == 1
```

Comment:
This conditional scans through the whole processor set twice, once to get the length, and once to construct the iterator.

Reply:
Consider:
```rust
let only_self = [self.vp.vp_index().index()].into_iter().eq(processor_set);
```